### DISCUSSION TITLE
Double-click a node in the Scene Tree to show or hide children

### DISCUSSION THREAD

**Discussion by samuel-walker** [godotengine/godot-proposals#8953] [Editor]
Hello. This is a pretty basic suggestion, and I'm not sure if it's already been discussed. A quick search didn't reveal anything.

I often work on Godot on a laptop with a trackpad. I find the small expander arrows next to nodes with children in the Scene Tree are a bit cumbersome to click on because they are small.

![expander-arrow](https://github.com/godotengine/godot-proposals/assets/31997383/3d09f111-dc01-4e88-bb25-a2b195fa426c)

I found myself today trying to instead double-click the node name or icon to expand/collapse the list of children. It doesn't appear there is any double-click functionality on nodes right now. I would propose adding a behavior to expand or collapse the children of the node once double-clicked. The double-clickable area is already pretty well defined by the area highlighted when the node is selected in the tree:

![interaction-area](https://github.com/godotengine/godot-proposals/assets/31997383/0743cdc9-5caa-428b-9a7f-6ad6ea6fd090)

If expand/collapse is not supported, I would also consider the following behaviors useful options for double-click behavior, again to avoid having to click small icons:

- open scene (if it's an instanced scene)
- open script (if it has an attached script)

But I think these options are worse, as in my projects at least, it is more likely a node has a child than is an instanced scene or has an attached script.

Finally, for accessibility reasons, it might also be wise to add a keyboard option for this behavior. You can navigate the scene tree and select nodes with the up/down arrow keys currently. Perhaps the Enter key would be the best option for matching the proposed double-click behavior, expanding/collapsing the selected node.

I imagine this proposal might affect all instances of trees in the editor, but I think this kind of behavior is appropriate for any tree UI object.

Thanks for considering this proposal.

**Comment by Calinou** [comment]
> It doesn't appear there is any double-click functionality on nodes right now

Double-clicking a node will show the rename form (if you click slowly the second time), or it will focus on the node in the 2D/3D editor (if you double-click quickly).

> Finally, for accessibility reasons, it might also be wise to add a keyboard option for this behavior. 

I believe <kbd>Right arrow</kbd> already expands a section in a tree and <kbd>Left arrow</kbd> collapses it. These shortcuts are standard in tree-style controls in most operating systems.

**Comment by samuel-walker** [comment]
Whoops! I am embarrassed. You are correct. I somehow missed both of those
things. Those make sense as default behaviours.
